src/components/Carousel.tsx

```diff
@@ -1,26 +1,103 @@
 import React from 'react';
 import './Carousel.scss';
+import classNames from 'classnames';
 
-const Carousel: React.FC = () => (
-  <div className="Carousel">
-    <ul className="Carousel__list">
-      <li>
-        <img src="./img/1.png" alt="1" />
-      </li>
-      <li>
-        <img src="./img/1.png" alt="2" />
-      </li>
-      <li>
-        <img src="./img/1.png" alt="3" />
-      </li>
-      <li>
-        <img src="./img/1.png" alt="4" />
-      </li>
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
+type Props = {
+  images: string[];
+  step: number;
+  frameSize: number;
+  itemWidth: number;
+  animationDuration: number;
+  infinite?: boolean;
+};
+
+const Carousel: React.FC<Props> = ({
+  images,
+  step,
+  frameSize,
+  itemWidth,
+  animationDuration,
+  infinite,
+}) => {
+  const [slide, setSlide] = React.useState(0);
+
+  const totalItems = images.length;
+  const slideItems = Math.max(Math.ceil((totalItems - frameSize) / step), 0);
+
+  const nextSlide = infinite || slide < slideItems;
+  const previousSlide = infinite || slide > 0;
+
+  const containerStyle = React.useMemo(
+    () => ({
+      width: `${itemWidth * frameSize}px`,
+    }),
+    [itemWidth, frameSize],
+  );
+
+  const listStyle = React.useMemo(() => {
+    const maxTranslateX = (totalItems - frameSize) * itemWidth;
+    const translateX = Math.min(slide * itemWidth * step, maxTranslateX);
+
+    return {
+      transform: `translateX(-${translateX}px)`,
+      transition: `transform ${animationDuration}ms ease-in-out`,
+    };
+  }, [totalItems, frameSize, itemWidth, slide, step, animationDuration]);
+
+  const handleNextClick = () => {
+    if (nextSlide) {
+      setSlide(currentSlide => (currentSlide + 1) % (slideItems + 1));
+    }
+  };
+
+  const handlePrevClick = () => {
+    if (previousSlide) {
+      setSlide(currentSlide =>
+        currentSlide === 0 ? slideItems : currentSlide - 1,
```

Comment:
```suggestion
        currentSlide === 0 ? slideItems : currentSlide - step,
```
logic is not clear to me: you should set correct slide taking into account `step` when click and do not use it when calculate translateX

Now, when I select step = 5, scroll to the end and then click `prev` button it scrolls 2 items instead of 5

Reply:
you're right, thanks